### PR TITLE
Ship vs ship revival

### PR DESCRIPTION
**Pull request by Tsunamico**
# About the pull request

This mnay not be Creedence Clearwater, but I'm damn sure it's a revival!

Starts a bit more of work into the Ship vs Ship systems already implemented!

# Explain why it's good for the game

Ship vs Ship makes Shipside people go yay.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: Added 3 new weapons to smack the ship with.
del: Nuked the Red Alert restriction that was on the General Quarters code for good.
refactor: Made the General Quarters call itself a standalone proc.
admin: Adds a panel to check the ship's health for Staff.
/:cl:

### CODE REVIEW

File: code/_globalvars/global_lists.dm [hunk 1]
```diff
@@ -15,6 +15,20 @@ GLOBAL_LIST_EMPTY(larva_burst_by_hive)
...
+"times_hit_railgun" = 0,
+"times_hit_odc" = 0,
+"times_hit_aaboiler" = 0,
+))
+
+//Ship's HP! ShipvsShip baybeee
+GLOBAL_LIST_INIT(ship_health_vars, list(
+"ship_hull_health" = 100,
+"ship_systems_health" = 100,
+))
```
Comment (Zonespace27) [inline code/_globalvars/global_lists.dm:30]: i feel like the ship being a datum is better

File: code/modules/admin/verbs/adminpanelweapons.dm [hunk 2]
```diff
@@ -97,6 +97,66 @@
 						addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(weaponhits), 2, picked_atom, point_defense), hiteta SECONDS)
 						message_admins("[key_name_admin(src)] Fired a single Railgun Slug at the Almayer at a random location, [picked_atom], with the possibility of missing as [point_defense]")
 
+		if("Orb. Def. Cannon")
+			if(exactplace == TRUE)
```
Comment (Zonespace27) [inline code/modules/admin/verbs/adminpanelweapons.dm:101]: you can just do `if(exactplace)` for truthy and `if(!exactplace)` for falsey

File: code/modules/admin/verbs/adminpanelweapons.dm [hunk 3]
```diff
@@ -97,6 +97,66 @@
...
+				shipwide_ai_announcement("DANGER: ORBITAL DEFENSES ENGAGING, INCOMING SHOT. BRACE, BRACE, BRACE. ESTIMATED TIME: [hiteta] SECONDS." , MAIN_AI_SYSTEM, 'sound/effects/missile_warning.ogg')
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(weaponhits), 3, mob.loc, point_defense), hiteta SECONDS)
```
Comment (Zonespace27) [inline code/modules/admin/verbs/adminpanelweapons.dm:103]: do you know that the mob's loc will always be a turf?

File: code/modules/almayer/weaponhits.dm [hunk 4]
```diff
@@ -16,6 +22,7 @@
  * * salvo - identifies it as a salvo or not.
  */
 /proc/weaponhits(weaponused, location, point_defense = FALSE, salvo = FALSE)
+	var/logtotalhits
 
 
 	switch(weaponused)
```
Comment (Zonespace27) [inline code/modules/almayer/weaponhits.dm:28]: everything in this file (weapons, what they do for damage, etc.) should 100% be datums

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/_globalvars/global_lists.dm (hunk 1 above)
Comment (Drulikar) [reply code/_globalvars/global_lists.dm:30]: An example of this specific code being datumized was here (but unfinished) https://github.com/cmss13-devs/cmss13/pull/3355 / https://github.com/cmss13-devs/cmss13/pull/3647

File: code/modules/admin/verbs/adminpanelweapons.dm (hunk 3 above)
Comment (Tsunamico) [reply code/modules/admin/verbs/adminpanelweapons.dm:103]: Would be misuse if it wasn't, it's for an admin to aim it at somewhere in a ship (Which kinda most certainly gonna be a turf) and then firin' it

File: code/modules/admin/verbs/adminpanelweapons.dm (hunk 2 above)
Comment (Tsunamico) [reply code/modules/admin/verbs/adminpanelweapons.dm:101]: :true:

**Comment by morrowwolf** [comment]
Staled, re-open when review is complete.

**Pull request closed without merging**